src/login/logind-dbus.c

```diff
@@ -2309,6 +2309,24 @@ static int setup_wall_message_timer(Manager *m, sd_bus_message* message) {
         return 0;
 }
 
+static int cache_wakeup_count(void) {
+        static const char kernel_wakeup_count[] = "/sys/power/wakeup_count";
+        static const char systemd_wakeup_count[] = "/run/wakeup_count";
+        _cleanup_free_ char *buf = NULL;
+        size_t bufsize;
+        int r;
+
+        r = read_full_virtual_file(kernel_wakeup_count, &buf, &bufsize);
+        if (r < 0)
+                return log_debug_errno(r, "Unable to read %s: %m", kernel_wakeup_count);
+
+        r = write_string_file(systemd_wakeup_count, buf, WRITE_STRING_FILE_DISABLE_BUFFER);
```

Comment:
WRITE_STRING_FILE_DISABLE_BUFFER is only for sysfs/procfs writes. But /run is a regular fs (i.e. tmpfs), hence don't use that flag.

Consider specifying WRITE_STRING_FILE_ATOMIC though, to make the thing updated atomically. this matters, because the systemd-sleep code is running async to this, and we really want to avoid that half-written files are read by it.